apps/webapp/src/state/swap.ts

```diff
@@ -127,24 +115,25 @@ const assembleSwapRequest = async ({ assetIn, amount, assetOut }: SwapSlice) =>
   });
 };
 
-const getSwapCommitment = (transaction: Transaction): StateCommitment => {
-  const swap = transaction.body?.actions.find(action => action.action.case === 'swap')?.action
-    .value as Swap | undefined;
-  if (!swap) throw new Error('Swap action could not be found in transaction for swap claim');
-  if (!swap.body?.payload?.commitment)
-    throw new Error('Swap commitment could not be found in swap action');
-
-  return swap.body.payload.commitment;
-};
-
-const assembleSwapClaimRequest = (swapCommitment: StateCommitment) => {
-  return new TransactionPlannerRequest({
+export const issueSwapClaim = async (swapCommitment: StateCommitment) => {
+  const swapClaimReq = new TransactionPlannerRequest({
     swapClaims: [
       {
         swapCommitment,
       },
     ],
   });
+  const swapClaimTx = await planWitnessBuildBroadcast(swapClaimReq, {
```

Comment:
When rebasing, this will need to use the new helpers @turbocrime split up in the helpers file.